internal/dag/gatewayapi_processor.go

```diff
@@ -93,11 +93,12 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 	)
 	defer commit()
 
-	var gatewayNotReadyCondition *metav1.Condition
+	var gatewayNotProgrammedCondition *metav1.Condition
 
 	if !isAddressAssigned(p.source.gateway.Spec.Addresses, p.source.gateway.Status.Addresses) {
-		gatewayNotReadyCondition = &metav1.Condition{
-			Type:    string(gatewayapi_v1beta1.GatewayConditionReady),
+		// TODO(sk) resolve condition type-reason mismatch
```

Comment:
The "AddressNotAssigned" reason did not move over to the "Programmed" condition, it's still documented as being for the "Ready" condition (ref. https://github.com/kubernetes-sigs/gateway-api/blob/eb822e981bd8f7b4f52e89cafbe2d0397b5f01fb/apis/v1beta1/gateway_types.go#L643-L648), so need to figure out what to do here. It would be a little odd to set "Ready: false" for this scenario, but never set "Ready: true" (since we're not currently able to meet the requirements for the "Ready: true" condition).

Reply:
I'm feeling like this is probably fine as-is for now since it's not breaking any conformance tests, but possibly something to follow up on down the road.

Reply:
hm yeah this one is a little weird, I could go either way on whether it belongs with the Programmed or Ready condition, but seems find to leave as you have it here until an upstream issue/conformance test changes things

Reply:
:+1: yeah agree this one could potentially go either way. I'll raise an issue upstream for it.